### PR TITLE
Update go version to 1.23.1

### LINKED ISSUE #2268: Update to go 1.23.1

### PR DESCRIPTION
**Pull request by dotNomad**
Used

```
go mod edit -go=1.23.1
```

## Intent

Resolves #2268

## Type of Change

- - [ ] Bug Fix 
- - [ ] New Feature 
- - [ ] Breaking Change 
- - [ ] Documentation 
- - [ ] Refactor 
- - [x] Tooling

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -20,13 +18,10 @@ require (
...
 )
 
-require golang.org/x/crypto v0.26.0 // indirect
-
 require (
 	github.com/alessio/shellescape v1.4.2 // indirect
 	github.com/danieljoos/wincred v1.2.1 // indirect
 	github.com/godbus/dbus/v5 v5.1.0 // indirect
-	google.golang.org/protobuf v1.34.2 // indirect  DON'T REMOVE, this is here to force a more recent version
 )
```
Comment (sagerb) [inline go.mod:25]: Any concern about the line above? ^^^ (`// indirect DON'T REMOVE, this is here to force a more recent version`)

File: go.sum [hunk 2]
```diff
@@ -64,24 +64,16 @@ github.com/stretchr/testify v1.9.0/go.mod h1:r2ic/lqez/lEtzL7wO/rwa5dbSLXVDPFyf8
...
-golang.org/x/sys v0.20.0/go.mod h1:/VUhepiaJMQUp4+oa/7Zr1D23ma6VTLIYjOOTFZPUcA=
 golang.org/x/sys v0.23.0 h1:YfKFowiIMvtgl1UERQoTPPToxltDeZfbj4H7dVUCwmM=
 golang.org/x/sys v0.23.0/go.mod h1:/VUhepiaJMQUp4+oa/7Zr1D23ma6VTLIYjOOTFZPUcA=
 golang.org/x/text v0.3.0/go.mod h1:NqM8EUOU14njkJ3fqMW+pc6Ldnwhi/IjpwHt7yyuwOQ=
-golang.org/x/text v0.15.0 h1:h1V/4gjBv8v9cjcR6+AR5+/cIYK5N/WAgiv4xlsEtAk=
-golang.org/x/text v0.15.0/go.mod h1:18ZOQIKpY8NJVqYksKHtTdi31H5itFRjB5/qKTNYzSU=
 golang.org/x/text v0.17.0 h1:XtiM5bkSOt+ewxlOE/aE/AKEHibwj/6gvWMl9Rsh0Qc=
 golang.org/x/text v0.17.0/go.mod h1:BuEKDfySbSR4drPmRPG/7iBdf8hvFMuRexcpahXilzY=
-google.golang.org/protobuf v1.34.2 h1:6xV6lTsCfpGD21XK49h7MhtcApnLqkfYgPcdHftf6hg=
-google.golang.org/protobuf v1.34.2/go.mod h1:qYOHts0dSfpeUzUFpOMr/WGzszTmLH+DiWniOlNbLDw=
```
Comment (sagerb) [inline go.sum:84]: I think my previous comment would want this dependency back in?

**Review by sagerb — COMMENTED** [review]
Had a concern regarding what was lost in the upgrade. Do we need to worry?

**Comment by dotNomad** [comment]
> Had a concern regarding what was lost in the upgrade.

I re-did this commit without the `go tidy` so the indirect requires should no longer be removed.

**Comment by dotNomad** [comment]
Immediately got an error when trying to build with just the version upgrade:
```
go: updates to go.mod needed, disabled by -mod=vendor
	(Go version in go.mod is at least 1.14 and vendor directory exists.)
	to update it:
	go mod tidy
```

So re-added the `go mod tidy`.

@mmarchetti would you mind taking a look, and letting me know if I'm missing something in our Go version upgrade flow?

**Comment by mmarchetti** [comment]
`go mod tidy` removes what it thinks are unused packages. We're overriding the version of protobuf because `github.com/spf13/afero`'s `go.mod` depends on a vulnerable version. This isn't the first vulnerable dependency afero has, and hasn't mitigated (there have only been 2 PRs merged this year in that repo). We might consider seeing if there's a fork that is better maintained, or fork it ourselves and updated the dependencies there.

**Comment by dotNomad** [comment]
Updated with a commit from @mmarchetti that updated the `go.mod` off of `main` and ran `go mod vendor`

**Review by sagerb — APPROVED** [review]
Upgraded to go 1.23.1 within VSCode and build runs successfully.

**Pull request merged**